tests/serverpod_test_server/test_integration/database_operations/transactions/transaction_isolatation_level_test.dart

```diff
@@ -0,0 +1,292 @@
+import 'dart:async';
+
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+void main() async {
+  var session = await IntegrationTestServer().session();
+
+  tearDown(() async {
+    await SimpleData.db.deleteWhere(session, where: (t) => Constant.bool(true));
+  });
+
+  group(
+      'Given read committed transaction isolation level and single row in database',
+      () {
+    var settings = TransactionSettings(
+      isolationLevel: IsolationLevel.readCommitted,
+    );
+
+    test(
+        'when row is modified after first statement in transaction'
+        'then transaction observes the updated value.', () async {
+      var testData = await SimpleData.db.insertRow(
+        session,
+        SimpleData(num: 1),
+      );
+      var c1 = Completer();
+      var c2 = Completer();
+      var transactionFuture = session.db.transaction(
+        (t) async {
+          await SimpleData.db.findById(
+            session,
+            testData.id!,
+            transaction: t,
+          );
+
+          c1.complete();
+          await c2.future;
+
+          return await SimpleData.db.findById(
+            session,
+            testData.id!,
+            transaction: t,
+          );
+        },
+        settings: settings,
+      );
+
+      await c1.future;
+      await SimpleData.db.updateRow(
+        session,
+        testData.copyWith(num: 2),
+      );
+      c2.complete();
+
+      var transactionResult = await transactionFuture;
+      expect(transactionResult?.num, 2);
+    });
+  });
+
+  group('Given repeatable read transaction isolation level', () {
+    var settings = TransactionSettings(
+      isolationLevel: IsolationLevel.repeatableRead,
+    );
+
+    test(
+        'when row is modified after first statement in transaction'
+        'then transaction does NOT observe the updated value.', () async {
+      var testData = await SimpleData.db.insertRow(
+        session,
+        SimpleData(num: 1),
+      );
+
+      var c1 = Completer();
+      var c2 = Completer();
+      var transactionFuture = session.db.transaction(
+        (t) async {
+          await SimpleData.db.findById(
+            session,
+            testData.id!,
+            transaction: t,
+          );
+
+          c1.complete();
+          await c2.future;
+
+          return await SimpleData.db.findById(
+            session,
+            testData.id!,
+            transaction: t,
+          );
+        },
+        settings: settings,
+      );
+
+      await c1.future;
+      await SimpleData.db.updateRow(
+        session,
+        testData.copyWith(num: 2),
+      );
+      c2.complete();
+
+      var result = await transactionFuture;
+      expect(result?.num, 1);
+    });
+
+    test(
+        'when read row is concurrently modified by other transaction'
```

Comment:
Question: Should there be an additional test for the isolation level Repeatable Read where updates are made to the row in which case it should throw?

Reply:
I wanted to stay out from testing the actual behavior of these things and focus on us setting the right configuration in the database. That is why I have developed the tests so that a set is unique for each configuration flag.

I don't think we need that test to be honest, we should be able to trust postgres to behave correctly.

Reply:
That's a good point!